packages/react-recommendations/src/useRecommendations.ts

```diff
@@ -63,6 +64,14 @@ export function useRecommendations<TObject extends ProductBaseRecord>(
       .then((record) => {
         const recommendations = record.recommendations ?? [];
 
+        if (
+          !props.searchClient.transporter.userAgent.value.includes(
+            `js-recommendations (${version})`
+          )
+        ) {
+          props.searchClient.addAlgoliaAgent('react-recommendations', version);
+        }
```

Comment:
We don't need to add the UA at every query, but we can rather add it every time the search client changes–in an effect with `props.searchClient` as dependencies.

Reply:
Done in https://github.com/algolia/react-recommendations/pull/7/commits/5d2720b33ba359eadf4c477dde72d4b2617302d9